lib/Runtime/Library/TypedArray.cpp

```diff
@@ -606,15 +606,39 @@ namespace Js
             object;
     };
 
-    BOOL TypedArrayBase::HasProperty(PropertyId propertyId)
+    BOOL TypedArrayBase::HasOwnProperty(PropertyId propertyId)
+    {
+        return JavascriptConversion::PropertyQueryFlagsToBoolean(HasPropertyQuery(propertyId));
+    }
+
+    inline BOOL TypedArrayBase::CanonicalNumericIndexString(PropertyId propertyId, ScriptContext *scriptContext)
+    {
+        PropertyString *propertyString = scriptContext->GetPropertyString(propertyId);
+        return CanonicalNumericIndexString(propertyString, scriptContext);
+    }
+
+    inline BOOL TypedArrayBase::CanonicalNumericIndexString(JavascriptString *propertyString, ScriptContext *scriptContext)
+    {
+        double result;
+        return JavascriptConversion::CanonicalNumericIndexString(propertyString, &result, scriptContext);
+    }
+
+    PropertyQueryFlags TypedArrayBase::HasPropertyQuery(PropertyId propertyId)
     {
         uint32 index = 0;
-        if (GetScriptContext()->IsNumericPropertyId(propertyId, &index) && (index < this->GetLength()))
+        ScriptContext *scriptContext = GetScriptContext();
+        if (scriptContext->IsNumericPropertyId(propertyId, &index))
         {
             // All the slots within the length of the array are valid.
-            return true;
+            return index < this->GetLength() ? Property_Found : Property_NotFound_NoProto;
```

Comment:
Being integer is guaranteed by IsNumericPropertyId(). Again I understand it may be confusing, but it has been around for long enough and entangled with names like "isNumeric" or "GetNumericValue", therefore I'd like to change them consistently, if at all, and probably get buy-in from 2/3 of the team. Leaving it alone for now.

---
In reply to: [106299416](https://github.com/Microsoft/ChakraCore/pull/2640#discussion_r106299416) [](ancestors = 106299416)